firebase-app-distribution/src/test/java/com/google/firebase/app/distribution/FirebaseAppDistributionTest.java

```diff
@@ -393,6 +394,59 @@ public void updateToNewRelease_whenCheckForUpdateFails_updateAppNotCalled() {
     assertEquals(FirebaseAppDistributionException.Status.NETWORK_FAILURE, e.getErrorCode());
   }
 
+  @Test
+  public void updateToNewRelease_whenTesterIsSignedIn_doesNotOpenDialog() {
+    when(mockSignInStorage.getSignInStatus()).thenReturn(true);
+    when(mockNewReleaseFetcher.checkForNewRelease())
+        .thenReturn(
+            Tasks.forException(
+                new FirebaseAppDistributionException(
+                    Constants.ErrorMessages.NETWORK_ERROR,
+                    FirebaseAppDistributionException.Status.NETWORK_FAILURE)));
+
+    firebaseAppDistribution.updateIfNewReleaseAvailable();
+
+    assertNull(ShadowAlertDialog.getLatestAlertDialog());
+  }
+
+  @Test
+  public void signInTester_whenDialogDismissed_taskFails() {
+    when(mockSignInStorage.getSignInStatus()).thenReturn(false);
+    Task updateTask = firebaseAppDistribution.updateIfNewReleaseAvailable();
+
+    AlertDialog dialog = verifySignInAlertDialog();
+    dialog.getButton(AlertDialog.BUTTON_NEGATIVE).performClick(); // dismiss dialog
+
+    assertFalse(updateTask.isSuccessful());
```

Comment:
Its only repeated once so its not necessary, but we could DRY up this code by putting this in another method to shorten this method and assertTaskFailure

Reply:
If you want to DRY this up, see [my PR](https://github.com/firebase/firebase-android-sdk/pull/3263/files#diff-691d14245f79be92331994b42451dda4fe8bd5fa297bf081604b7eda370b4fa4R177-R190) for a possible approach.

Reply:
SGTM!

Reply:
Addressed this feedback in https://github.com/firebase/firebase-android-sdk/pull/3286